opensquirrel/writer/writer.py

```diff
@@ -13,10 +13,11 @@ def __init__(self, register_manager: RegisterManager) -> None:
         self.register_manager = register_manager
         qubit_register_size = self.register_manager.get_qubit_register_size()
         qubit_register_name = self.register_manager.get_qubit_register_name()
+
         bit_register_size = self.register_manager.get_bit_register_size()
         bit_register_name = self.register_manager.get_bit_register_name()
-        self.output = "version 3.0\n\n{}\n{}\n".format(
-            f"qubit[{qubit_register_size}] {qubit_register_name}",
+        self.output = "version 3.0\n".format(
+            "\n{}\n{}\n"+f"qubit[{qubit_register_size}] {qubit_register_name}" if qubit_register_size > 0 else "",
             f"bit[{bit_register_size}] {bit_register_name}\n" if bit_register_size > 0 else "",
```

Comment:
```suggestion
        self.output = "version 3.0{}{}{}{}\n".format(
            "\n\n" if qubit_register_size > 0 or bit_register_size > 0,
            f"qubit[{qubit_register_size}] {qubit_register_name}" if qubit_register_size > 0 else "",
            "\n" if qubit_register_size > 0 and bit_register_size > 0 else "",
            f"bit[{bit_register_size}] {bit_register_name}" if bit_register_size > 0 else "",
```

I would suggest this. A `version 3.0` line followed by 3 fields:
1. If we only have qubits, we'll write: version, blank line, qubits.
2. If we qubits and bits, we'll write: version, blank line, qubits, bits.
3. If we only have bits, we'll write: version, blank line, bits.
4. If we don't have neither qubits nor bits, we'll write: version.

All the options end with a newline.

Reply:
Hi Robert, what would be the best place to test this? I would say test_integration.py, or do you think another place would be better?

Reply:
I would probably add the tests to `test/parser/libqasm/test_libqasm.py`. There you have already a `test_simple`. You could add, for example, `test_simplest_qubits_but_no_bits`, `test_simplest_qubits_and_bits`, `test_simplest_no_qubits_but_bits`, and `test_simplest_no_qubits_and_no_bits`.